### PR TITLE
Feature/import parameter name input

### PR DESCRIPTION
**Pull request by Rdmkr**
Parameter name / descriptions input in CSV import dialog, as requested in this ticket:
https://sourceforge.net/p/minsky/ravel/463/

This version only has the extra input fields and no changes to the layout of the dialog yet. See ticket on further suggestions on improvements to the CSV import dialog's layout.

This change is [Reviewable](https://reviewable.io/reviews/highperformancecoder/minsky/507)

### CODE REVIEW

**Comment by highperformancecoder** [comment]
I won't merge this right now - firstly it is mixed up with your work on the URL downloader, so a merge up from master branch is required. Secondly, since there's no actions attached to the new input fields, its not a good look to go to release with this. I'd be happy to merge this into develop branch, though.

**Comment by Rdmkr** [comment]
I've added the layout changes to this branch.

"firstly it is mixed up with your work on the URL downloader, so a merge up from master branch is require"
I started this branch from that other branch, so there would not be conflicts. Just merged master into this branch without any problems.
" since there's no actions attached to the new input fields"
Clarify this please?

"I'd be happy to merge this into develop branch, though."
Acceptable to me, although I think our first users will like to see the new layout, which I could put in a different branch if required.

Please give confirmation on the preferred approach

File: gui-js/libs/ui-components/src/lib/import-csv/import-csv.component.html [hunk 1]
```diff
@@ -1,199 +1,242 @@
...
+          </mat-autocomplete>
+        </span>
 
-      <span>
-        <label for="horizontal-dimension"
-              title="Name of the collection of data columns.">Horizontal Dimension</label>
-        <div class="wedge-container">
+        <span>
+          <label for="decimal-separator" title="Separator of units and
+          fraction part of numbers: '.' for UK/US data, '.' for European data.">Decimal Separator</label>
```
Comment (highperformancecoder) [inline gui-js/libs/ui-components/src/lib/import-csv/import-csv.component.html:91]: Don't you mean ',' for European data?

File: gui-js/libs/ui-components/src/lib/import-csv/import-csv.component.html [hunk 2]
```diff
@@ -1,199 +1,242 @@
...
-      <ng-container [formGroup]="horizontalDimension">
         <span>
-          <label for="type" title="Type of horizontal dimension.">Type</label>
-          <select id="type" formControlName="type">
-            <option value="string">string</option>
-            <option value="value">value</option>
-            <option value="time">time</option>
-          </select>
+          <label for="escape" title="Character used to embed special
+          characetrs in strings, such as quotes.">Escape</label>
```
Comment (highperformancecoder) [inline gui-js/libs/ui-components/src/lib/import-csv/import-csv.component.html:107]: Typo 'characters'

**Comment by Rdmkr** [comment]
I'll be happy to correct those last two errors, but they were already there before I started work on this

**Review by highperformancecoder — APPROVED** [review]
I fixed the typos in the merge.

**Comment by highperformancecoder** [comment]
On Tue, Jun 11, 2024 at 01:32:44AM -0700, Rdmkr wrote:
> I'll be happy to correct those last two errors, but they were already there
> before I started work on this

Fair enough. I fixed them during the merge anyway :). Looking good, by
the way!


-- 

----------------------------------------------------------------------------
Dr Russell Standish                    Phone 0425 253119 (mobile)
Principal, High Performance Coders     ***@***.***
                      http://www.hpcoders.com.au
----------------------------------------------------------------------------

**Pull request merged**